machina/traj/epi_functional.py

```diff
@@ -124,6 +157,44 @@ def compute_advs(data, gamma, lam):
     return data
 
 
+def compute_hs(data, func, hs_name='hs', input_acs=False):
+    """
+    Computing Hidden State of RNN Cell.
+
+    Parameters
+    ----------
+    data : Traj
+    func : 
+        Any function. for example pols, vf and qf.
+
+    Returns
+    -------
+    data : Traj
+    """
+    epis = data.current_epis
+    func.reset()
+    with torch.no_grad():
+        for epi in epis:
+            obs = torch.tensor(
+                epi['obs'], dtype=torch.float, device=get_device()).unsqueeze(1)
+            time_seq = obs.size()[0]
+            if input_acs:
+                acs = torch.tensor(
+                    epi['acs'], dtype=torch.float, device=get_device()).unsqueeze(1)
+                hs_seq = [func(obs[i:i+1], acs[i:i+1])[-1]['hs']
+                          for i in range(time_seq)]
```

Comment:
`[hs for hs in func(obs, acs)['hs']]` may be faster.

Reply:
We have to get hs at all time steps. If we input obs and acs of all time steps at once, we get only last hs.

Reply:
I got it.